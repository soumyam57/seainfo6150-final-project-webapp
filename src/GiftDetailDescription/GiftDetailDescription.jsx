import React from 'react';
import PropTypes from "prop-types";
import styles from "./GiftDetailDescription.module.css";

const GiftDetailDescription= props => {
    return (
        <div className={styles.description}>
            <p>Price: {props.detail.price}</p><br />
            <p>Status: {props.detail.status}</p><br />
            <p>Category: {props.detail.category}</p><br />
            <p>Category Name: {props.detail.categoryid}</p><br />
            <p>Short Description: {props.detail.shortDesc}</p><br />
            <p>Description: {props.detail.description}</p><br />
        </div>
    )
}

GiftDetailDescription.propTypes = {
    detail: PropTypes.object.isRequired
};

export default GiftDetailDescription;