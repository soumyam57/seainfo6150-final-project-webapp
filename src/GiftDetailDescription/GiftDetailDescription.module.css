.description{
    padding: 1%;
    border-radius: 10%;
    width: 100%;
    height: 100%
}

