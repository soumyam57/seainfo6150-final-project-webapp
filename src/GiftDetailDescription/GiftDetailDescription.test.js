import React from "react";
import { render } from "@testing-library/react";
import GiftDetailDescription from "./GiftDetailDescription.jsx";

const giftdetails={
          "price":"$1000",
          "title":"Couple Rings",
          "shortDesc":"CHuir Wedding Couple Rings for Him and Her, Adjustable Celtic Knot Promise Ring Set",
          "category":"wedding",
          "categoryid":"couplerings",
          "img":
          {
             "url":"https://images-na.ssl-images-amazon.com/images/I/51mfwq0NDEL._AC_UX522_.jpg",
             "height":400,
             "width":500
          },
          "status": "In Stock",
          "description": ["MATERIALS Made of 30% Silver Alloy, White Gold Plating, AAA Grade Cubic Zirconia Stone RINGS SIZE", 
                         "Adjustable design, and don't worry about the size. Adjustable size range (US size) : Women: 4-8, Men: 5-10. Gift for her and him or yourself",
                         "PACKAGE INCLUDES 1 x Pair of Couple Ring Set (one for Her, the other for Him) in a simple pakage"]
}

it("renders the GiftDetailDescription component correctly", () => {
    const { container } = render(
        <GiftDetailDescription
            detail={giftdetails}
        />
    )
});