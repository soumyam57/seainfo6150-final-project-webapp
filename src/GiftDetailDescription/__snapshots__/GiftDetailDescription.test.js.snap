// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the GiftDetailDescription component correctly 1`] = `
<div>
  <div
    class="description"
  >
    <p
      class="title"
    >
      Price: 
      $1000
    </p>
    <br />
    <p
      class="title"
    >
      Status: 
      In Stock
    </p>
    <br />
    <p>
      Category: 
      wedding
    </p>
    <br />
    <p>
      Category Name: 
      couplerings
    </p>
    <br />
    <p>
      Short Description: 
      CHuir Wedding Couple Rings for Him and Her, Adjustable Celtic Knot Promise Ring Set
    </p>
    <br />
    <p>
      Description: 
      MATERIALS Made of 30% Silver Alloy, White Gold Plating, AAA Grade Cubic Zirconia Stone RINGS SIZE
      Adjustable design, and don't worry about the size. Adjustable size range (US size) : Women: 4-8, Men: 5-10. Gift for her and him or yourself
      PACKAGE INCLUDES 1 x Pair of Couple Ring Set (one for Her, the other for Him) in a simple pakage
    </p>
    <br />
  </div>
</div>
`;
